### PR TITLE
Error: Self signed certificate in chain #1432

### PR DESCRIPTION
**Pull request by sameer-haider**
Added certificate error handler in main.ts which prevents default rejections of self-signed certificates. Added unit testing, and updated mock files so that tests run smoothly.

### CODE REVIEW

File: packages/main/src/main.ts [hunk 1]
```diff
@@ -92,6 +92,13 @@ app.on('ready', async () => {
   }
 });
 
+export function handleCertificateError(event, webContents, url, error, certificate, callback) {
+  event.preventDefault();
```
Comment (nukeop) [inline packages/main/src/main.ts:96]: Can you log that this occurred using the logger?

**Comment by sameer-haider** [comment]
Okay I added the logger, looks like the Windows test didn't run properly before, any clue how to debug that?

**Comment by nukeop** [comment]
You can ignore the windows build, that's a broken pipeline that I'll have to fix later.

**Comment by klapperkopp** [comment]
@nukeop this does not seem fixed, I am still getting the error in latest version for MacOS aarch if youtube is the source. Other sources sometimes work.

`Command failed: ERROR: Unable to download API page: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1081) (caused by CertificateVerifyError('[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1081)')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U`

`resolveCandidatesForTrack error: yt-dlp failed: ERROR: Unable to download API page: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1081) (caused by CertificateVerifyError('[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1081)')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
`

Seems that's more related to this bug: https://github.com/yt-dlp/yt-dlp/issues/3868

**Comment by nukeop** [comment]
Ok, IMO, if this comes from yt-dlp, it's a yt-dlp issue (which Nuclear uses as an external binary). No way for me to fix it here.

**Pull request merged**